spec/gilded_rose_spec.js

```diff
@@ -1,7 +1,166 @@
+  //    2 when there are 10 days or less and by 3 when there are 5 days or less but quality drops to 0 after the concert
+  // We have recently signed a supplier of conjured items. This requires an update to our system:
+  //    "Conjured" items degrade in quality twice as fast as normal items
+
+
+  // Normal Item
+
+  items=[]
+
+  it("Normal Item: should decrement both sell in and quality after running update_quality", function() {
+    var currSize=items.length;
+    console.log('-->'+currSize)
+    items.push(new Item('Test', 2, 10));
 
-  it("should do something", function() {
     update_quality();
+
+    expect(items[currSize].sell_in).toEqual(1);
+    expect(items[currSize].quality).toEqual(9);
   });
 
+  it("should test for for sell_in less than zero, quality degrades twice as fast", function () {
+    var currSize=items.length;
+    items.push(new Item('Test for negative sell in', -2, 5));
+
+    update_quality();
+
+    expect(items[currSize].sell_in).toEqual(-3);
+    expect(items[currSize].quality).toEqual(3);
+  })
+
+  it("Quality of an item is never negative", function () {
+    // (item name, sell_in, quality)
+    var currSize=items.length;
+    items.push(new Item('Zero Quality Item', 2, 0));
+
+    update_quality();
+
+    expect(items[currSize].quality).toEqual(0);
+  })
+
+  it("Quality of an item is never negative when quality is decreasing at twice the rate", function () {
+    // (item name, sell_in, quality)
+    var currSize=items.length;
+    items.push(new Item('Zero Quality Item And Zero Date', -2, 0));
+
+    update_quality();
+
+    expect(items[currSize].quality).toEqual(0);
+  })
+
+  it("Aged Brie actually increases in quality the older it gets", function () {
+    var currSize=items.length;
+    // (item name, sell_in, quality)
+    items.push(new Item('Aged Brie', 2, 2));
+    var previousQuality = items[currSize].quality;
+
+    update_quality();
+
+    //new quality rating should be 3
+    expect(items[currSize].quality).toBeGreaterThan(previousQuality);
+  })
+
+  it("The quality of an item is never more than 50", function() {
+    var currSize = items.length;
+    // (item name, sell_in, quality)
+    items.push(new Item('Aged Brie', 2, 50));
+    var previousQuality = items[currSize].quality;
+
+    update_quality();
+    expect(items[currSize].quality).toBeLessThanOrEqual(50);
+  })
+
+  it("Sulfuras, being a legendary item, never has to be sold nor does it decrease in quality", function() {
+    var currSize = items.length;
+    // (item name, sell_in, quality)
+    items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 48));
+    update_quality();
+    expect(items[currSize].quality).toEqual(48);
+    expect(items[currSize].sell_in).toEqual(0);
+  })
+
+  it("Backstage passes quality increases by 2 when there are 10 days or less", function() {
+    var testArray = [];
```

Comment:
`testArray` is an unnecessary variable. This could be written as `items = [new Item('Backstage passes to a TAFKAL80ETC concert', 10, 20)]`